scripts/testy_launch.py

```diff
@@ -117,140 +117,147 @@ def snapshot_exists(snapshot_id):
+    max_retries = 30
+    sleep_time = 10
+    retry = 0
 
-    while num_retry < max_retries:
-        result = local(f"aws ec2 describe-instance-status \
-            --instance-ids {instance_id} \
-            --query 'InstanceStatuses[*].InstanceState.Code' \
-            --output text", hide=True, warn=True)
-        if result.stderr:
-            raise Exit(result.stderr)
-        state = result.stdout.strip()
-        if state != expected_state:
-            num_retry += 1
-            time.sleep(10)
-        else:
-            break
-    else:
-        raise Exit(f"Timeout: the instance {instance_id} is not running.")
+    timer = cycle(['\\', '|', '/', '—'])
+    msg = f"Waiting for the status check to complete. This may take several minutes ..."
+    print("", end=f"{msg}\r", flush=True)
 
-def wait_instance_status_check_ok(instance_id):
-    print(f"Waiting for the EC2 instance '{instance_id}' to pass the status check ...")
-    max_retries = 20
-    num_retry = 0
+    while retry < max_retries:
 
-    while num_retry < max_retries:
         result = local(f"aws ec2 describe-instance-status \
             --instance-ids {instance_id} \
             --query 'InstanceStatuses[*].[InstanceStatus.Status,SystemStatus.Status]' \
-            --output text", hide=True, warn=True)
+            --output json", hide=True, warn=True)
         if result.stderr:
             raise Exit(result.stderr)
-        checks = result.stdout.strip().split()
-
-        while True:
-            if not len(checks):
-                return
-            if checks[0] != 'ok':
-                break
-            checks.pop(0)
-        num_retry += 1
-        time.sleep(10)
+
+        status = json.loads(result.stdout)
+        if (len(status) == 0 or len(status[0]) != 2):
+            continue
+        if status[0][0] == "ok" and status[0][1] == "ok":
+            print("", end=f"{msg} Success!\n", flush=True)
+            break
+        retry += 1
+        for _ in range (sleep_time):
+            print("", end=f"{msg} {next(timer)}\r", flush=True)
+            time.sleep(1)
+
     else:
-        raise Exit(f"Timeout: the instance {instance_id} has not passed the status check yet.")
+        print("", end=f"{msg} Timed out.\n", flush=True)
+        raise Exit(f"The status check failed to complete successfully after "
+            f"{max_retries*sleep_time} seconds. Please check the AWS console.")
 
 # Launch an AWS instance given a distro.
-# This function returns a dictionary with a 'status' field that is set to 0 only when the instance
-# has been launched successfully. In that case, the dictionary contains a 'user' and 'host' fields.
-# Upon failure, the 'status' field is not 0 and the 'msg' field contains the error message.
-def testy_launch(distro):
+# This function returns a dictionary with a 'status' field that is set to 0 when the instance
+# is launched successfully. In the successful case, the dictionary also contains information
```

Comment:
Should we name the keys that the dictionnary contains?

Reply:
I updated this comment in [175bc3c](https://github.com/wiredtiger/testy/pull/50/commits/175bc3c6927b8d3eacdad0b7dbfdd805afbaf36c). I am generally not in favor of referring to specific variable names in comments but rather use comments to describe the intent of the function and choosing variable names wisely so that the code is "self-documenting".